QEfficient/utils/generate_inputs.py

```diff
@@ -111,7 +117,12 @@ def update_pytorch_inputs(self, inputs, pt_outputs):
             updated_inputs["batch_index"] = torch.arange(self.full_batch_size).view(-1, 1)
 
         else:
-            updated_inputs["input_ids"] = pt_outputs["logits"].argmax(-1).reshape(-1, 1)
+            if self.num_speculative_tokens:
+                # assume spec decoding logits
+                input_ids = pt_outputs["logits"][:, -1].argmax(-1).reshape(-1, 1)
+            else:
+                input_ids = pt_outputs["logits"].argmax(-1).reshape(-1, 1)
+            pt_outputs["input_ids"] = input_ids

```

Comment:
Could you please confirm whether it should be pt_outputs["input_ids"] = input_ids or updated_inputs["input_ids"] = input_ids